### ISSUE TITLE
Chore: compact GitHub Pull Request URLs 

### ISSUE THREAD

**Issue by nelsonic** [dwyl/link#7]
As noted by @LuchoTurtle in https://github.com/dwyl/link/pull/5#issuecomment-1662141984 the current implementation of at `compact_github_url/1` [`lib/link.ex#L52-L85`](https://github.com/dwyl/link/blob/ac42d24137f201820d72df029bb410d482197033/lib/link.ex#L52-L85) 
does not correctly handle `Pull Request` URLs. 

https://amemo.fly.dev
```sh
https://github.com/dwyl/link
https://github.com/dwyl/link/pull/5  
https://github.com/dwyl/link/pull/5#pullrequestreview-1558913764  
```
![image](https://github.com/dwyl/link/assets/194400/6ac1c8ac-5740-49c5-97b1-cd508fa9f1a5)

This despite the tests passing!! 
https://github.com/dwyl/link/blob/ac42d24137f201820d72df029bb410d482197033/test/link_regex_test.exs#L85-L98

Need help figuring out _why_ this works in our tests but not in the demo app. 🤷‍♂️

**Comment by nelsonic** [comment]
OK, I _think_ I've understood the problem ... 🕵️‍♂️ 🔍 💡 🤦‍♂️ 

If the links in `<textarea>` are all _different_ then the `replace` function works fine:

```md
https://github.com/dwyl/mvp spacing text 
https://github.com/dwyl/link/pull/4 more text  
https://github.com/dwyl/link/pull/5#pullrequestreview-1558913764 
```

![image](https://github.com/dwyl/link/assets/194400/2732d214-bb77-43cc-a497-12cf9772d65f)

But if all the links are to the _same_ repo e.g: https://github.com/dwyl/link
Then the it messes up because the `replace` is replacing the links in a loop!

![image](https://github.com/dwyl/link/assets/194400/73123535-da85-4000-82d6-dd11020d831f)

I haven't _yet_ figured out a way _around_ this. 💭 
But I'm leaning toward a cheeky hack using whitespace. 🤔

**Comment by nelsonic** [comment]
I've added "hacky" code that I'm not proud of but that passes the tests. 🧑‍💻 ✅

**Comment by nelsonic** [comment]
Working on `localhost`: 

```md
https://github.com/dwyl/link spacing text  
https://github.com/dwyl/link/pull/5 more text    
https://github.com/dwyl/link/pull/5#pullrequestreview-1558913764  
more random text and a markdown link
[link](https://github.com/dwyl/link)
```

![image](https://github.com/dwyl/link/assets/194400/30fc1623-9e32-4908-938a-c649c70f0059)

That's me hovering over the link to show that the target is still correct ✅

**Comment by nelsonic** [comment]
Works on https://amemo.fly.dev/ live demo version. ✅ 

```md
https://github.com/dwyl/link spacing text  
https://github.com/dwyl/link/pull/5 more text    
https://github.com/dwyl/link/pull/5#pullrequestreview-1558913764  
more random text and a markdown link
[link](https://github.com/dwyl/link)
```

![image](https://github.com/dwyl/link/assets/194400/8c87dc68-07eb-4ea8-ade8-1d4a0e417de4)

**Comment by nelsonic** [comment]
Included in #5
